src/data-types/char.ts

```diff
@@ -41,22 +48,24 @@ module.exports = {
     }
   },
 
-  writeTypeInfo: function(buffer, parameter) {
+  writeTypeInfo: function(buffer, parameter: ParameterData<Buffer | null>) {
```

Comment:
Here as well, let's just use `any` as the generic parameter:

```suggestion
  writeTypeInfo: function(buffer, parameter: ParameterData<any>) {
```